### PR TITLE
Get Prior Living Situation logic correct; Add support for initial values

### PR DESCRIPTION
**Pull request by gigxz**
* Remove `NOT_EXISTS` operator in favor of `EXISTS` with `answerBoolean` value (like FHIR)
* Add `IN` operator and `answerCodes` value. This lets us do something like `the answer to question A is in list [A,B,C], AND some other thing is true` which previously wouldn't be possible within the `ANY`/`ALL` enable-behavior framework.
* Add `ENABLED` operator. **EXISTS** means that the item should have a non-null _answer_, **ENABLED** just means that the item should be enabled. This is slightly more work at eval time, because we are evaluating the same enableWhen repeatedly, but we can optimize that later. I think this pattern will make it easier to write these definitions. One bad thing is that it could cause and infinite loop at eval time if there is a dependency cycle, but we could add a check for that when validating the definition.
* Add `initial` type (matching FHIR again here). Not used yet.
* Corresponding PR https://github.com/greenriver/hmis-frontend/pull/65 has evaluation tests!

### CODE REVIEW

File: drivers/hmis/lib/form_data/assessments/dummy_intake_assessment.json [hunk 1]
```diff
@@ -41,58 +41,71 @@
...
+                "LOS_90_DAYS_OR_MORE_BUT_LESS_THAN_ONE_YEAR",
+                "LOS_ONE_YEAR_OR_LONGER"
               ]
+            }
+          ]
+        },
+        {
+          "type": "DISPLAY",
+          "link_id": "3.917.B",
+          "text": "Client stayed 7+ days in a transitional or permanent housing situation. This is considered a \"break\" according to the HUD definition of chronic homelessness. Stopping data collection for 3.917 Prior Living Situation.",
```
Comment (gigxz) [inline drivers/hmis/lib/form_data/assessments/dummy_intake_assessment.json:69]: This is now simply a conditionally displayed message. I think this pattern could be really useful.

File: drivers/hmis/lib/form_data/assessments/dummy_intake_assessment.json [hunk 2]
```diff
@@ -41,58 +41,71 @@
...
+              "question": "3.917.A",
+              "operator": "ENABLED",
+              "answer_boolean": false
+            },
+            {
+              "question": "3.917.B",
+              "operator": "ENABLED",
+              "answer_boolean": false
+            }
+          ]
```
Comment (gigxz) [inline drivers/hmis/lib/form_data/assessments/dummy_intake_assessment.json:107]: This is the implementation of "Stop data collection for 3.917 Prior Living Situation."

This is saying: if either of the "stopping data collection..." display items are enabled, hide this.

File: drivers/hmis/lib/form_data/assessments/dummy_intake_assessment.json [hunk 3]
```diff
@@ -41,58 +41,71 @@
...
               ]
             }
           ]
         },
+        {
+          "type": "BOOLEAN",
+          "link_id": "3.917.C",
+          "text": "On the night before, did the client stay on the streets, ES or SH?",
+          "project_types_excluded": ["ES", "SERVICES_ONLY", "SH"],
+          "enable_behavior": "ALL",
```
Comment (eanders) [inline drivers/hmis/lib/form_data/assessments/dummy_intake_assessment.json:95]: Shouldn't this be 'ANY', or is it that when this is disabled (none are true) then disable something else?  I'd love a demo of this.

**Review by eanders — APPROVED** [review]

**Comment by gigxz** [comment]
Can we merge this @eanders?

**Pull request merged**